File: my-solutions/0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if(len(nums) == 0):
            return 0
        num_set = set(nums)
        max_len = 1
        for num in nums:
            curr_len = 0
            if num + 1 in num_set:
                continue
            elif (num - 1 in num_set):
                temp = num
                while(temp in num_set):
                    temp = temp - 1
                    curr_len += 1
            max_len = max(curr_len, max_len)
        return max_len

File: my-solutions/0128-longest-consecutive-sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // convert the array of numbers to a set.
        unordered_set<int> s(nums.begin(), nums.end());

        int max_len = 0;
        // find sequence starting numbers.
        for(int i = 0; i < size(nums); i++) {
            int curr_len = 1;
            if(s.find(nums[i] - 1) == s.end()) {
                int j = 1;
                while(s.find(nums[i] + j++) != s.end())
                    curr_len++;
            }
            cout << curr_len << endl;
            if(max_len < curr_len) {
                max_len = curr_len;
            }
        }

        return max_len;
    }
};
